src/threading.c

```diff
@@ -215,6 +215,12 @@ void ti_threadfun(void *arg)
     while (ta->state == TI_THREAD_INIT)
         cpu_pause();
     cpu_lfence();
+
+    // Assuming the functions called below doesn't contain unprotected GC
+    // critical region. In general, the following part of this function
+    // shouldn't call any managed code without calling `jl_gc_managed_enter`
+    // first.
+    jl_gc_state_set_and_save(3, 0);
```

Comment:
I originally meant unmanaged code but I don't think that distinction is important.